File: projects/10/ruby/birds.rb
# birds.rb | Tests out the menagerie of bird classes.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: ZeAi Sun
# Date: Mar 27, 2023
######################################################

require './Bird.rb'
require './FlyingBird.rb'
require './WalkingBird.rb'
require './Duck.rb'
require './Goose.rb'
require './Owl.rb'
require './Penguin.rb'
require './Ostrich.rb'
require './Kiwi.rb'

puts "\nWelcome to the Bird Park!\n\n"

bird1 = Duck.new "Donald"
bird1.print

bird2 = Goose.new "Mother"
bird2.print

bird3 = Owl.new "Woodsey"
bird3.print

bird4 = Penguin.new "Peter"
bird4.print

bird5 = Ostrich.new "Orville"
bird5.print

bird6 = Kiwi.new "Katherine"
bird6.print

puts "\n\n"

File: projects/10/java/Birds.java
/* Birds.java illustrates inheritance and polymorphism.
 *
 * Begun by: Joel Adams, for CS 214 Lab 10 at Calvin College.
 * Completed by: ZeAi Sun
 * Date: Mar 25, 2023
 *********************************************************/

public class Birds {
  
  public static void main(String[] args) {
      System.out.println("\nWelcome to the Bird Park!\n");

      Duck bird1 = new Duck("Donald");
      bird1.print();

      Goose bird2 = new Goose("Mother");
      bird2.print();

      Owl bird3 = new Owl("Woodsey");
      bird3.print();

      Penguin bird4 = new Penguin("Peter");
      bird4.print();

      Ostrich bird5 = new Ostrich("Orville");
      bird5.print();

      Kiwi bird6 = new Kiwi("Katherine");
      bird6.print();

      System.out.println();
    }
}

